docs/user/input.md

```diff
@@ -377,9 +380,46 @@ so only the `flow` parameter is required. The presence of sediment concentration
 requires the `sediment` parameter, as the presence of salinity concentrations
 requires the `salinity` parameter.
 
+## `checkpoint`
+
+```yaml
+checkpoint:
+  format: hdf5
+  interval: 100
+```
+
+The `checkpoint` section contains fields that specify whether and how RDycore
+writes checkpoint files, which can be used to restart simulations using
+parameters in the `restart` section.
+
+* `format`: the format of the checkpoint files to be written. This can be either
+  `binary` (the default) or `hdf5`.
+* `interval`: the number of time steps taken between writing checkpoint files.
+  This must be a positive integer.
+* `prefix`: an optional prefix for checkpoint files. If omitted, the prefix for
+  checkpoint files is the prefix of the YAML input file name for the simulation.
+
+The name of a checkpoint file written at time step `N` is `<prefix>-N.<format>`,
```

Comment:
In E3SM, we have checkpoint/restart files named `<prefix>.<component>.r.YYYY-MM-DD-HHMMSS.nc`. So, in this PR we should change that filename to be `<prefix>.rdycore.r.N.<format>`. Later, we should allow an external driver (aka E3SM) to dictate the suffix of the restart file so that it would be something like `<prefix>.rdycore.r.<suffix>.<format>`.